### DISCUSSION TITLE
HIDDEN values

### DISCUSSION THREAD

**Discussion by Rustimation** [philippoo66/optolink-splitter#41] [Q&A] [answered]
My Parameter file DP_VScotHO1_200_11.txt contains lines like this: 
_Externe Betriebsarten-umschaltung M2 (271) [ExtBetriebsartenumschaltung_M2~0x3549 (Byte)] **HIDDEN**:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="5 A1/M1 + M3")_
What does HIDDEN mean in this context and how would I read these values. Or should I just ignore that?

**Comment by philippoo66** [comment]
HIDDEN means that those values do not appear in Vitosoft if the conditions (not) meat. In this case you might ignore them because they are useless then.

**Comment by Rustimation** [comment]
If I understand this correctly, when I'm only using the Optolink-Splitter, I can ignore the hidden values, right?

**Reply by philippoo66** [reply]
not completely correct. you have to check under wich conditions (meantioned) those values are hidden. If your installation meats the 'hidden condition', then you ignore the values. If not, the values might be meaningful (carry 'real' data).

**Reply by philippoo66** [reply]
example: 
```
- Betriebsdaten A1 (22374) HIDDEN:(00_WWS="7 M2 + M3" OR 00_WWS="8 M2 + M3 + WW" OR 00_WWS="3 M2" OR 00_WWS="4 M2 + WW")
    - Betriebsart A1M1 (92) [BedienteilBA_GWGA1~0x2323 (Byte)]
...
```

If you have mixer circuit/s M2, M3 then e.g. `Betriebsart A1M1 (92) [BedienteilBA_GWGA1~0x2323 (Byte)]` is hidden an you use 0x350x instead.

this is 'written' in 
```
- Heizkreis M2 (22391) HIDDEN:(00_WWS="1 A1" OR 00_WWS="2 A1 + WW")
    - Aktuelle Betriebsart M2 (709) [HK_AktuelleBetriebsartM2~0x3500 (Byte)]
...
```

**Comment by Rustimation** [comment]
Okay, I'll try to digest this and may come back in case questions remain. Grazie mille
Chris

**Reply by philippoo66** [reply]
probably 00_WWS is indicated by 0x7700 ('Heizkreis-**W**arm**w**asser**s**chema')....

for values 1 and 2 -> 0x3500 is hidden, with values 7, 8, 3, 4 -> 0x2323 is hidden.

**Reply by Rustimation** [reply]
Thank you very much... but as I lack the Viessmann Specialist knowledge and not using Vitosoft or Viessdata, I couldn't possibly know how to find out which "Betriebsart" or "Warmwasserschema" are present in my Vitodens 200. Bohemian villages to me... 
Would installing your Viessdata tool help understand things better?

**Reply by Rustimation** [reply]
My problem is, how do I get from `HIDDEN:(00_WWS="1 A1" OR 00_WWS="2 A1 + WW")` to `0x7700 ('Heizkreis-Warmwasserschema')`
IMHO that is something you actually have to "know" as it is not really obvious.
Nonetheless I have made some progress:
I checked 0x7700, the result is "8" --> Heizkreis A1M1 says it's hidden `00_WWS="8 M2 + M3 + WW"` although I don't know what M2 + M3 + WW really means in context with HIDDEN. 
For Heizkreis 2 and 3 it the HIDDEN Rule doesn't catch so there is data in the respective sections for the heating circuits. 
Ein richtiges Suchspiel!

**Reply by philippoo66** [reply]
@Rustimation  sorry for my late reply! jap, "Ein richtiges Suchspiel!" it truely is. As Optolink was never meant to get used by the customer/device owner, it is not documentated by Viessmann and probably not realy 'consistent'. 

> Would installing your Viessdata tool help understand things better?

No. Viessdata wasn't originally made by me and bears lots of further 'inconsistencies' especially if not used with a Vitodens 300 HO2B in 'simple circuit setup'.

`00_WWS="8 M2 + M3 + WW"` would mean that you have two mixer circuits and hot water conditioning setup (do you have?). In this case the `Heizkreis A1M1` 'parameters' often do not apply. 

In some cases Vitosoft might help to 'answer some questions' but you need to know how to use it (SQL queries etc). I never installed it since it brings a lot of 'old time' servers and engines with it which may not be compatible with a current windows setup. I heard that after installing it, the system was 'deeply corrupted' and other applications stopped working. So _if_ you consider to install, do not do on your productive system.

But @FransOv got it running and is quite familiar with the 'usage'. So often he can help you if you tell him what you need, on the other hand he does not know your setup offhand...

**Comment by FransOv** [comment, marked as answer]
I just added the output from the query on the Vitosoft database for the  Vitodens 200 mit HO1B  to the Excel in the Wiki https://github.com/philippoo66/optolink-splitter/wiki/310-Parameter-Addresses on tab VScotHO1_200_11.

**Reply by Rustimation** [reply]
Hello Frans,
thank you very much. I will try this out. The good thing ist that your xls contains more values than the text file DP_VScotHO1_200_11.txt
Not that I could make head or tail of it. E.g. 0xA152 is some kind of relay status flag, however it has 16 or so different meanings. Whatever "nvo" means... 
rgds Chris

**Reply by FransOv** [reply]
When there are many descriptions for one code, this often means that this a compound parameter containing multiple info's. As this 0xA152 is 2 bytes with 16 info's, it might well be a 16 bit bitmap. However the query does not preserve order so the sequence is uncertain. I have also sometimes seen a 2 byte temperature and a one byte sensor status that had the same code and could only be read as a 3 byte parameter.

**Reply by philippoo66** [reply]
Hi Frans @FransOv
could you pls do me a favor and generate the Excel sheet for CU401B_S (F8: 204D) ?!
thank you very much!
Phil

**Reply by FransOv** [reply]
@philippoo66 
Hi Phil,

I added the list to the Excel.

Frans

**Reply by philippoo66** [reply]
thank you very much @FransOv !! also for doing so prompty!

Unfortunately the addresses do not match the 204D Vitocal 200-S device we are currently working on. Outside temperature is 0x0101 as stated on https://github.com/openv/openv/wiki/Adressen . Every data set derived for CU401B_S from Vitosoft gives 0x01C1 (SQL aswell as xml). 

Do you have any idea?

ps. just got the idea that perhaps it is one of these 3-byte temperatures (incl status). we will try tonight... does it return 3,...,04 when trying to read these 3-byte vals with 2 bytes?
